### PR TITLE
Escape both quotes and backslashes for sql insert

### PR DESCRIPTION
**Pull request by bethanyaconnor**
The rollup script was failing on levels with already escaped quotes, such as [this level](https://github.com/code-dot-org/code-dot-org/blob/4ec1e11deaa5f459bd6c4b482f68adc5ed1fca59/dashboard/config/scripts/csa_u2_assessment_q14.multi#L4). This PR also escapes backslashes.

I also added some rescues so that we attempt to generate all the tables even if one of them fails.

It may be easier to view this diff with [whitespace ignored](https://github.com/code-dot-org/code-dot-org/pull/42723/files?diff=split&w=1).

## Links

[Honeybadger error](https://app.honeybadger.io/projects/45435/faults/51801288)
[jira](https://codedotorg.atlassian.net/browse/PLAT-1332)

## Testing story

tested locally

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/cron/create_rollup_tables [hunk 1]
```diff
@@ -16,7 +17,7 @@ TIME_NOW = DateTime.now.strftime('%F %T').freeze
 #   quotations escaped. Returns nil on nil input.
 def sanitize_string_for_db(unsanitized)
   return nil if unsanitized.nil?
-  unsanitized.gsub(/[\r\n]+/, '').gsub(/"/, '\"')
+  unsanitized.gsub(/[\r\n]+/, '').gsub(/\\|"/, '\\\\\\0')
```
Comment (bethanyaconnor) [inline bin/cron/create_rollup_tables]: I got this lovely substitution from https://apidock.com/ruby/String/gsub#787-Replacing-with-and-match. I'm open to better options here but had trouble coming up with anything.

**Comment by davidsbailey** [comment]
A few thoughts here:
1. csa_u2_assessment_q14.multi appears to have smart quotes in it. not sure if that is relevant somehow?
2. would `.dump`work?
```
[development] dashboard > s = '`System.out.println(You say \“ goodbye \”, \n I say \“ hello \” .);`'
=> "`System.out.println(You say \\“ goodbye \\”, \\n I say \\“ hello \\” .);`"
[development] dashboard > puts s.dump
"`System.out.println(You say \\\u201C goodbye \\\u201D, \\n I say \\\u201C hello \\\u201D .);`"
```

**Comment by bethanyaconnor** [comment]
> have smart quotes in it

I don't think it does. I think sql is just treating them as normal quotes. The error I'm seeing without any changes is ```Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'goodbye \\", \n I say \\" hello \\" .);`", ```

> would .dump work?

I thought I tried that but let me check what the issue was

**Review by davidsbailey — APPROVED** [review]

**Comment by bethanyaconnor** [comment]
Alright, I got `.dump` to work and I think that's a better solution. The issues were coming from spots where there wasn't a value but the script run successfully for me now.

I think there is an issue where we're missing the question from contained levels sometimes because they're stored as a different property. I'd rather look into that as a follow-up though.

**Comment by davidsbailey** [comment]
> Alright, I got `.dump` to work and I think that's a better solution. The issues were coming from spots where there wasn't a value but the script run successfully for me now.

Ah, great, I'm glad this worked!
 
> I think there is an issue where we're missing the question from contained levels sometimes because they're stored as a different property. I'd rather look into that as a follow-up though.

sounds good to me.

**Pull request merged**